server/vulnerabilities/cpe.go

```diff
@@ -139,67 +144,312 @@ func cleanAppName(appName string) string {
 	return strings.TrimSuffix(appName, ".app")
 }
 
-var onlyAlphaNumeric = regexp.MustCompile("[^a-zA-Z0-9]+")
-
-func CPEFromSoftware(db *sqlx.DB, software *fleet.Software) (string, error) {
-	targetSW := ""
-	switch software.Source {
-	case "apps":
-		targetSW = "macos"
-	case "python_packages":
-		targetSW = "python"
-	case "chrome_extensions":
-		targetSW = "chrome"
-	case "firefox_addons":
-		targetSW = "firefox"
-	case "safari_extensions":
-		targetSW = "safari"
-	case "deb_packages":
-	case "portage_packages":
-	case "rpm_packages":
-	case "npm_packages":
-		targetSW = `"node.js"`
-	case "atom_packages":
-	case "programs":
-		targetSW = `"windows*"`
-	case "ie_extensions":
-	case "chocolatey_packages":
+var onlyAlphaNumeric = regexp.MustCompile(`[^a-zA-Z0-9]+`)
+
+// sanitizeMatch sanitizes the search string for sqlite fts queries. Replaces all special characters with spaces.
+func santizeMatch(s string) string {
+	return onlyAlphaNumeric.ReplaceAllString(s, " ")
+}
+
+var sanitizeVersionRe = regexp.MustCompile(`[^a-zA-Z0-9_-]+`)
+
+// sanitizeVersion attempts to sanitize versions and attempt to make it dot separated.
+// Eg Zoom reports version as "5.11.1 (8356)". In the NVD CPE dictionary it should be 5.11.1.8356.
+func sanitizeVersion(version string) string {
+	parts := onlyAlphaNumeric.Split(version, -1)
+	return strings.Trim(strings.Join(parts, "."), ".")
+}
+
+// TODO: add more vendors
+var macOSVendors = map[string]string{
+	"com.postmanlabs.mac":           "getpostman",
+	"org.virtualbox.app.VirtualBox": "oracle",
+}
+
+const cpeTranslationsFilename = "cpe_translations.json"
+
+func loadCPETranslations(path string) (CPETranslations, error) {
+	f, err := os.Open(path)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+
+	var translations CPETranslations
+	if err := json.NewDecoder(f).Decode(&translations); err != nil {
+		return nil, fmt.Errorf("decode json: %w", err)
+	}
+
+	return translations, nil
+}
+
+func DownloadCPETranslations(vulnPath string, client *http.Client) error {
+	ghClient := github.NewClient(client)
+
+	opts := &github.RepositoryContentGetOptions{
```

Comment:
Currently depends on this branch https://github.com/fleetdm/nvd/tree/michal-6628-macos-vuln. This should be removed before merging.

I am not sure this is the best way to fetch and keep the cpe translations up to date, but it's similar to what we are doing for `oval_sources.json`. Should we change this to add these to the release assets for fleetdm/nvd ie https://github.com/fleetdm/nvd/releases/tag/8078aad03ca2d810?

Reply:
What are the advantages of having this file as part of the release assets?

Reply:
Main reason is consistency. If someone tries to find which versions of `cpe.sqlite`, 'oval_sources.json`, and `cpe_translations.json` were downloaded and from where, they have to look in Releases and the repo files.

Reply:
Sounds good to me - I'm all in favor of doing whatever is more intuitive.